src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -334,7 +334,11 @@ JNI_METHOD(void, shutdownSubscriptions)(JNIEnv * env, jobject self, jlong handle
     chip::DeviceLayer::StackLock lock;
 
     DeviceProxy * device = reinterpret_cast<DeviceProxy *>(devicePtr);
-    device->ShutdownSubscriptions();
+
+    //
+    // TODO: Consult with austin if we can delete this function entirely in the PR.
+    //
+    // device->ShutdownSubscriptions();
```

Comment:
@austinh0 do you guys this method? Can I delete this entire function if possible?

Reply:
Medium term, we need to shift this logic to having it own and manage the allocation and lifetime of the ReadClient object directly.

Reply:
Without this call Android CHIPTool crashes after exiting to the main screen from a fragment that enabled subscriptions, because subsequent reports are trying to update GUI elements that are in "incorrect" state. What mechanism can an application use now to shutdown a started subscription that is no longer needed?

Reply:
You should be managing the instantiation and lifetime management of the ReadClient directly instead of going through the convenience `SubscribeAttribute` methods on `CHIPCluster.h`.

I'll put these back for now, but will file an issue to track the transition.

Reply:
#13163